src/wesnoth.cpp

```diff
@@ -267,10 +267,46 @@ static int handle_validate_command(const std::string& file, abstract_validator&
 }
 
 /** Process commandline-arguments */
-static int process_command_args(const commandline_options& cmdline_opts)
+static int process_command_args(commandline_options& cmdline_opts)
 {
 	// Options that don't change behavior based on any others should be checked alphabetically below.
 
+	if(cmdline_opts.no_log_sanitize) {
+		lg::set_log_sanitize(false);
+	}
+
+	// decide whether to redirect output to a file or not
+	if(cmdline_opts.log_to_file) {
+		cmdline_opts.final_log_redirect_to_file = true;
+	} else if(cmdline_opts.no_log_to_file) {
+		cmdline_opts.final_log_redirect_to_file = false;
+	} else {
+		// write_to_log_file means that writing to the log file will be done, if true.
+		// if false, output will be written to the terminal
+		// on windows, if wesnoth was not started from a console, then it will allocate one
+		cmdline_opts.final_log_redirect_to_file = !getenv("WESNOTH_NO_LOG_FILE")
+		// command line options that imply not redirecting output to a log file
+		// Some switches force a Windows console to be attached to the process even
+		// if Wesnoth is an IMAGE_SUBSYSTEM_WINDOWS_GUI executable because they
+		// turn it into a CLI application. Also, --no-log-to-file in particular attaches
+		// a console to a regular GUI game session.
+			&& !cmdline_opts.data_path
+			&& !cmdline_opts.help
+			&& !cmdline_opts.logdomains
+			&& !cmdline_opts.nogui
+			&& !cmdline_opts.report
+			&& !cmdline_opts.simple_version
+			&& !cmdline_opts.userdata_path
+			&& !cmdline_opts.version
+			&& !cmdline_opts.do_diff
+			&& !cmdline_opts.do_patch
+			&& !cmdline_opts.preprocess
+			&& !cmdline_opts.render_image
+			&& !cmdline_opts.screenshot
+			&& !cmdline_opts.headless_unit_test
+			&& !cmdline_opts.any_validation_option();
```

Comment:
This is clearly incorrect. The `--validate-addon` option shouldn't turn it into a CLI program, and I think `--validate-core` probably shouldn't either. These are "validate as you play" options. Only `--validate` and `--validate-schema` are intended for CLI use. (The fifth one, `--use-schema`, is only an auxiliary to `--validate` and meaningless without it.)